documentation/src/main/asciidoc/userguide/chapters/domain/embeddables.adoc

```diff
@@ -197,4 +197,99 @@ This usage is covered in detail in <<chapters/domain/identifiers.adoc#identifier
 [IMPORTANT]
 ====
 Embeddable types that are used as collection entries, map keys or entity type identifiers cannot include their own collection mappings.
-====
\ No newline at end of file
+====
+
+[[embeddable-Target]]
+==== `@Target` mapping
+
+The https://docs.jboss.org/hibernate/orm/{majorMinorVersion}/javadocs/org/hibernate/annotations/Target.html[`@Target`] annotation is used to specify the implementation class of a given association that is mapped via an interface.
+The
+http://docs.oracle.com/javaee/7/api/javax/persistence/ManyToOne.html[`@ManyToOne`],
+http://docs.oracle.com/javaee/7/api/javax/persistence/OneToOne.html[`@OneToOne`],
+http://docs.oracle.com/javaee/7/api/javax/persistence/OneToMany.html[`@OneToMany`], and
+http://docs.oracle.com/javaee/7/api/javax/persistence/ManyToMany.html[`@ManyToMany`]
+feature a http://docs.oracle.com/javaee/7/api/javax/persistence/ManyToOne.html#targetEntity--[`targetEntity`] attribute to specify the actual class of the entity association when an interface is used for the mapping.
+
+The http://docs.oracle.com/javaee/7/api/javax/persistence/ElementCollection.html[`@ElementCollection`] association has a http://docs.oracle.com/javaee/7/api/javax/persistence/ElementCollection.html#targetClass--[`targetClass`] attribute for the same purpose.
```

Comment:
It's not related to this patch but we should probably update all these EE 7 links to EE 8 links considering we are targeting JPA 2 now.

WDYT?

Reply:
Makes sense. I'll add a new issue for it.